Sources/VLCSettingsController.m

```diff
@@ -199,9 +201,43 @@ - (UIView *)settingsViewController:(id<IASKViewController>)settingsViewControlle
     return header;
 }
 
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    [tableView deselectRowAtIndexPath:indexPath animated:YES];
+    IASKSpecifier *specifier = [self.settingsReader specifierForIndexPath:indexPath];
+
+    if ([specifier.type isEqualToString: kIASKPSMultiValueSpecifier]) {
+        [self displayActionSheetFor:specifier];
+    } else {
+        [super tableView:tableView didSelectRowAtIndexPath:indexPath];
+    }
+}
+
 - (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section
 {
     return nil;
 }
 
+- (void)displayActionSheetFor:(IASKSpecifier *)specifier
+{
+    VLCActionSheet *actionSheet = [[VLCActionSheet alloc] init];
+    actionSheet.modalPresentationStyle = UIModalPresentationCustom;
+    [actionSheet.collectionView registerClass:[VLCSettingsSheetCell class] forCellWithReuseIdentifier:VLCSettingsSheetCell.identifier];
+    
+    specifierManager = [[VLCSettingsSpecifierManager alloc] initWithSpecifier:specifier settingsReader:self.settingsReader settingsStore:self.settingsStore];
+    actionSheet.delegate = specifierManager;
+    actionSheet.dataSource = specifierManager;
+    
+    [actionSheet setActionWithClosure:^(id item) {
```

Comment:
can we use something else than id here?

Reply:
I don't think so because the closure is receiving bunch of different types as `item`. But even if we were able to type check `item` as something like `long`, InAppSettings doesn't provide a `set___` method for all types